### PR TITLE
Add support for Mode SixtyFive M65HA and M65HI

### PR DESCRIPTION
**Pull request by Gondolindrim**
This PR adds initial compatibility for Mode SixtyFive's M65HA and M65HI PCBs, respectively the ANSI and the ISO versions of its hotswap PCBs.

Both PCBs had two versions, a prototype (pre-Alpha) and a final run (Alpha); while the mass-production will ship with versions Alpha, pre-Alphas had a significant number of units made, hence their firmwares are listed for archiving purposes. Also release Alpha had matrix changes, hence why there are firmwares for both. Hence this PR adds four PCBs: M65HA pA and A, and M65HI pA and A.

The firmwares listed follow the conventions and suggestions by QMK reviewers in the soldered PCB revision (see https://github.com/qmk/qmk_firmware/pull/14027).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mode/m65ha_alpha/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Mode M65HA (pre-release Alpha version)
```
Comment (drashna) [inline keyboards/mode/m65ha_alpha/README.md:1]: Please rename the file to be lower case.

File: keyboards/mode/m65ha_alpha/rules.mk [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = STM32F401
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = full     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/mode/m65ha_alpha/rules.mk:7]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/mode/m65ha_alpha/rules.mk [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/mode/m65ha_alpha/rules.mk:18]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/xbows/knight/keymaps/via/config.h [hunk 4]
```diff
@@ -1,64 +1,64 @@
...
-#    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS      // Hue & value pulse away on the same column and row of a single key hit then fades value out
-#    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS // Hue & value pulse away on the same column and row of multiple key hits then fades value out
-//#    define DISABLE_RGB_MATRIX_SPLASH                    // Full gradient & value pulse away from a single key hit then fades value out
-#    define DISABLE_RGB_MATRIX_MULTISPLASH               // Full gradient & value pulse away from multiple key hits then fades value out
-#    define DISABLE_RGB_MATRIX_SOLID_SPLASH              // Hue & value pulse away from a single key hit then fades value out
-//#    define DISABLE_RGB_MATRIX_SOLID_MULTISPLASH         // Hue & value pulse away from multiple key hits then fades value out
-
-
-#endif
+/* Copyright 2021 Shulin Huang <mumu@x-bows.com>
```
[the remark this answers is not in the document]
Comment (Gondolindrim) [reply keyboards/xbows/knight/keymaps/via/config.h:1]: Why do you guys assume there's a reason for everything? Maybe I'm just stupid.

**Comment by Jaicob** [comment]
@fauxpark can you please give this PR another review or reassign someone who is available to?

File: keyboards/xbows/knight/keymaps/via/config.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (fauxpark) [reply keyboards/xbows/knight/keymaps/via/config.h:1]: This is still outstanding.

**Comment by Gondolindrim** [comment]
@fauxpark thanks for your patience -- I force-pushed the past commits to only include the M65H files.

File: keyboards/mode/m65hi_pre-alpha/m65hi_preAlpha.c [hunk 5]
```diff
@@ -0,0 +1,109 @@
+/*
```
Comment (fauxpark) [inline keyboards/mode/m65hi_pre-alpha/m65hi_preAlpha.c:1]: All file and folder names should be lowercase and not contain `-`s.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Gondolindrim** [comment]
In talks with Mode we decided to keep the pre-alpha firmwares in our forks and just submit the alpha final production files, because the configurator files were not checking out.

File: keyboards/mode/m65ha_alpha/m65ha_alpha.c [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+        eeconfig_update_user(tmp.raw);
+        uint32_t value = eeconfig_read_user();
+        if (value != tmp.raw) {
+            dprint("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n");
+            dprint("!! EEPROM readback mismatch!\n");
+            dprint("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n");
+        }
+    }
+}
```
Comment (fauxpark) [inline keyboards/mode/m65ha_alpha/m65ha_alpha.c:109]: What is this for?

**Review by fauxpark — APPROVED** [review]

**Pull request merged**